llvm/lib/SYCLLowerIR/LowerWGScope.cpp

```diff
@@ -726,29 +722,20 @@ static void sharePFWGPrivateObjects(Function &F, const Triple &TT) {
   Instruction &At = LeaderBB->back();
 
   for (auto &Arg : F.args()) {
-    Type *T;
-    LLVMContext &Ctx = At.getContext();
-    IRBuilder<> Builder(Ctx);
-    Builder.SetInsertPoint(&LeaderBB->front());
+    if (!Arg.hasByValAttr())
```

Comment:
Bit: we skip "this" because it is allocated in the proper AS by the FE, correct? Comment would be helpful for the reader.

Reply:
Right. I just reverted the changes from https://github.com/intel/llvm/pull/1455 and tried to re-implement it by fixing address space in clang instead.
Do you want me to comment that `this` points to the object in local address space, so we don't need a shadow copy for that argument?